tpm/ak.go

```diff
@@ -0,0 +1,204 @@
+package tpm
+
+import (
+	"context"
+	"crypto/rand"
+	"fmt"
+	"time"
+
+	"github.com/google/go-attestation/attest"
+
+	"go.step.sm/crypto/tpm/storage"
+)
+
+type AK struct {
+	Name      string
+	Data      []byte
+	CreatedAt time.Time
+
+	tpm *TPM
+}
+
+func (t *TPM) CreateAK(ctx context.Context, name string) (AK, error) {
+	result := AK{}
+	if err := t.Open(ctx); err != nil {
+		return result, fmt.Errorf("failed opening TPM: %w", err)
+	}
+	defer t.Close(ctx, false)
+
+	at, err := attest.OpenTPM(t.attestConfig)
+	if err != nil {
+		return result, fmt.Errorf("failed opening TPM: %w", err)
+	}
+	defer at.Close()
+
+	now := time.Now()
+
+	if name == "" {
+		// TODO: decouple the TPM key name from the name recorded in the storage? This might
+		// make it easier to work with the key names as a user; the TPM key name would be abstracted
+		// away. The key name in the storage can be different from the key stored with the key (which,
+		// to be far, isn't even used on Linux TPMs)
+		nameHex := make([]byte, 5)
+		if n, err := rand.Read(nameHex); err != nil || n != len(nameHex) {
+			return result, fmt.Errorf("rand.Read() failed with %d/%d bytes read and error: %w", n, len(nameHex), err)
```

Comment:
The number of bytes read aren't all that important here. A much simpler way to express this failure would be something like: 

```go
fmt.Errorf("failed reading from CSPRNG: %w", err")
```